### ISSUE TITLE
Why does it take too long for Onionshare to finally give me a URL to share?

### ISSUE THREAD

**Issue by delaRosa12345** [micahflee/onionshare#266]
Hi!

I think it's been past 5 mins since I uploaded the file. I've got a Tor Browser open already. I'm even only uploading a non high res jpeg :(
Why is this?

Thanks!

**Comment by micahflee** [comment]
This isn't related to the size of the files you're sharing. My best guess is it's related to using ephemeral hidden services instead of the old kind. I've also noticed that occasionally it takes forever to start the hidden service, but most of the time it works quickly. I've closed OnionShare and re-opened it, and the second time it works quickly. I'm not sure how to reproduce it consistently though.

Basically, OnionShare turns your computer into a web server, and makes the website accessible as a hidden service. You can be sharing a 5kb file or a 5gb file, and the only time difference will be in how long it takes to zip up the files. But after that, all of the waiting is just waiting for Tor to successfully propagate the new .onion.

**Comment by delaRosa12345** [comment]
Ah ok, got it.

**Comment by micahflee** [comment]
It's difficult to reproduce this reliably, and OnionShare does seem to always eventually work and return a .onion URL. So until there's a reliable way to test for this, I'm going to close this issue.
